api/v1/_uploadPublicFile/PKPUploadPublicFileHandler.inc.php

```diff
@@ -0,0 +1,189 @@
+	 */
+	private function getResponse($response) {
+		return $response->withHeader('Access-Control-Allow-Headers', 'Content-Type, X-Requested-With, X-PINGOTHER, X-File-Name, Cache-Control');
+	}
+
+	/**
+	 * Upload a requested file
+	 *
+	 * @param $slimRequest Request Slim request object
+	 * @param $response Response object
+	 * @param $args array arguments
+	 * @return Response
+	 */
+	public function uploadFile($slimRequest, $response, $args) {
+		$request = $this->getRequest();
+
+		if (empty($_FILES) || empty($_FILES['file'])) {
+			return $response->withStatus(400)->withJsonError('api.temporaryFiles.400.noUpload');
+		}
+
+		$siteDir = Core::getBaseDir() . '/' . Config::getVar('files', 'public_files_dir') . '/site';
+
+		if (!file_exists($siteDir) || !is_writeable($siteDir)) {
+			return $response->withStatus(500)->withJsonError('api.publicFiles.500.badFilesDir');
+		}
+		$userDir = $siteDir . '/images/' . $request->getUser()->getUsername();
+		$isUserAllowed = true;
+		$allowedDirSize = 25000 * 1024; // 25mb
+		$allowedFileTypes = ['gif', 'jpg', 'png'];
+
+		HookRegistry::call('API::uploadPublicFile::permissions', [
+			&$userDir,
+			&$isUserAllowed,
+			&$allowedDirSize,
+			&$allowedFileTypes,
+			$request,
+			$this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES),
+		]);
+
+		// Allow plugins to control who can upload files
+		if (!$isUserAllowed) {
+			return $response->withStatus(403)->withJsonError('api.publicFiles.403.unauthorized');
+		}
+
+		// Don't allow user to exceed the alotted space in their public directory
+		$currentSize = 0;
+		if ($allowedDirSize > 0 && file_exists($userDir)) {
+			foreach (new RecursiveIteratorIterator(new RecursiveDirectoryIterator($userDir, FilesystemIterator::SKIP_DOTS)) as $object) {
+				$currentSize += $object->getSize();
+			}
+		}
+		if (($currentSize + $_FILES['file']['size']) > $allowedDirSize) {
+			return $response->withStatus(413)->withJsonError('api.publicFiles.413.noDirSpace', [
+				'fileUploadSize' => ceil($_FILES['file']['size']/1024),
+				'dirSizeLeft' => ceil(($allowedDirSize - $currentSize)/1024),
+			]);
+		}
+
+		import('lib.pkp.classes.file.FileManager');
+		$fileManager = new FileManager();
+		$filename = $fileManager->getUploadedFileName('file');
+		$extension = end(explode('.', strtolower(trim($filename))));
+
+		// Only allow permitted file types
+		if (!in_array($extension, $allowedFileTypes)) {
+			return $response->withStatus(400)->withJsonError('api.publicFiles.400.extensionNotSupported', [
+				'fileTypes' => join(__('common.commaListSeparator'), $allowedFileTypes)
+			]);
+		}
+
+		// Perform additional checks on images
+		if (in_array($extension, ['gif', 'jpg', 'jpeg', 'png', 'jpe'])) {
+			if (getimagesize($_FILES['file']['tmp_name']) === false) {
+				return $response->withStatus(400)->withJsonError('api.publicFiles.400.invalidImage');
+			}
+			$extensionFromMimeType = $fileManager->getImageExtension(mime_content_type($_FILES['file']['tmp_name']));
```

Comment:
Use `PKPString::mime_content_type` instead of `mime_content_type` -- it supports fallbacks for systems without `mime_content_type` support.

Reply:
https://github.com/pkp/pkp-lib/commit/f86f3f89be82b0d1d3c54e3aefb7307fb66db31d